nixarr/default.nix

```diff
@@ -274,10 +287,10 @@ in {
 
             text = ''
               mkdir -p "$(dirname ${cfg.stateDir})"
-              if [ ! -f ${cfg.api-key-location} ]; then
-                openssl rand -hex 64 > ${cfg.api-key-location}
+              if [ ! -f ${cfg.api-key-location-internal} ]; then
+                openssl rand -hex 64 > ${cfg.api-key-location-internal}
               fi
-              chgrp media ${cfg.api-key-location}
+              chgrp media ${cfg.api-key-location-internal}
```

Comment:
Apologies, I think this still doesn't fix my problem with this line. Since `api-key-location-internal` takes the value of `api-key-location` if it is defined, this will still cause issues if `api-key-location` is readonly.

I do really think you just need to move that statement into the if block above though - if the api-key is provided by the user, it is up to them to make sure it has the correct owner and group membership

Reply:
Alright, I think this stylistic nitpicking is not worth holding up the PR any further.